include/fmt/format.h

```diff
@@ -1513,16 +1522,24 @@ template <typename Range> class basic_writer {
     struct num_writer {
       unsigned_type abs_value;
       int size;
+      const std::string& groups;
       char_type sep;
 
       template <typename It> void operator()(It&& it) const {
         basic_string_view<char_type> s(&sep, sep_size);
         // Index of a decimal digit with the least significant digit having
         // index 0.
         unsigned digit_index = 0;
+        std::string::const_iterator group = groups.cbegin();
         it = internal::format_decimal<char_type>(
-            it, abs_value, size, [s, &digit_index](char_type*& buffer) {
-              if (++digit_index % 3 != 0) return;
+            it, abs_value, size, [this, s, &group, &digit_index](char_type*& buffer) {
+              if (*group <= 0
+                || *group == CHAR_MAX
```

Comment:
I suggest using `max_value<char>()` instead of `CHAR_MAX` and removing the `climits` include.

Reply:
I've changed it to max_value<char>().